src/docfx/build/metadata/ContributionInfo.cs

```diff
@@ -200,5 +202,22 @@ private GitUserInfo ToGitUserInfo(UserProfile profile)
 
             return result;
         }
+
+        private async Task<UserProfile> GetUserProfile(GitCommit commit, Document document)
+        {
+            if (string.IsNullOrEmpty(commit.AuthorEmail))
+                return null;
+
+            var result = _userProfileCache.GetByUserEmail(commit.AuthorEmail);
+            if (result == null)
+            {
+                var (repo, _, _) = GetRepository(document);
+                var authorName = await _github.GetNameByCommit(repo.Owner, repo.Name, commit.Sha);
+                var profile = (await _github.GetUserProfileByName(authorName)).AddEmail(commit.AuthorEmail);
+                result = _userProfileCache.AddOrUpdate(authorName, profile, (k, v) => v.AddEmail(commit.AuthorEmail));
```

Comment:
>_userProfileCache.AddOrUpdate [](start = 25, length = 29)

This is not thread safe, per discussion we could update cache while loading the commits

Reply:
moved this logic into LoadCommits()

---
In reply to: [209483593](https://github.com/dotnet/docfx/pull/3176#discussion_r209483593) [](ancestors = 209483593)